### PR TITLE
[WiP] Elm 0.19

### LINKED ISSUE #202: Migrate to Elm 0.19
Because, why not? :-)

### PR DESCRIPTION
**Pull request by vjousse**
Fixes #202 

Still a work in progress. First step, make it compile, even if not half of the feature are present.
Second step, make it work as expected.

- [x] Make it compile with elm 0.19
- [x] Add useful date formatting back for `formatDate` function
- [x] Add scrolling behavior back to `src/Command.elm`
- [x] Update to `elm/http` 2.0.0
- [x] Add Keyboard behavior back in  `src/Subscription.elm` using https://package.elm-lang.org/packages/elm/browser/latest/Browser-Events
- [x] Fix `textContent` function in `src/View/Formatter.elm`
- [x] Add WebSocket support using ports
- [x] Change `tests` to work with latest `elm-test`
- [x] Replace `Autocomplete` with https://github.com/ContaSystemer/elm-menu/tree/1.0.2
- [x] Update `cleanErrors` in `src/Update/Error.elm`

### CODE REVIEW

File: elm.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "dependencies": {
+        "direct": {
+            "ContaSystemer/elm-menu": "1.0.2",
+            "NoRedInk/elm-json-decode-pipeline": "1.0.0",
+            "billstclair/elm-port-funnel": "1.2.0",
+            "billstclair/elm-websocket-client": "4.1.0",
+            "elm/browser": "1.0.2",
+            "elm/core": "1.0.2",
+            "elm/html": "1.0.0",
+            "elm/http": "1.0.0",
```
Comment (Natim) [inline elm.json]: Ok porting to 2.0.0 would be yet another story then :)

File: src/Mastodon/WebSocket.elm [hunk 2]
```diff
@@ -34,28 +36,31 @@ type alias WebSocketMessage =
...
-                        (replaceSlice "wss" 0 5 <| client.server ++ ApiUrl.streaming)
-                        [ ( "access_token", client.token )
-                        , ( "stream", type_ )
-                        ]
-            in
-                WebSocket.listen url message
-
-        Nothing ->
-            Sub.none
+    {-
```
Comment (Natim) [inline src/Mastodon/WebSocket.elm]: This is going to be turned into a port.

File: src/Util.elm [hunk 3]
```diff
@@ -14,17 +13,29 @@ acceptableAccounts query accounts =
...
-        [ _, authCode ] ->
-            Just authCode
+
+
+{-
+   TODO: refactor this code smell
+-}
+
+
+extractAuthCode : Url.Url -> Maybe String
```
Comment (Natim) [inline src/Util.elm]: We might want to use Url.Parser.Query there.

File: src/View/App.elm [hunk 4]
```diff
@@ -98,26 +104,36 @@ homepageView model =
...
+                Nothing ->
+                    authView model
+            , case model.viewer of
+                Just viewer ->
+                    viewerView viewer
+
+                Nothing ->
+                    text ""
+            , case model.confirm of
+                Nothing ->
```
Comment (Natim) [inline src/View/App.elm]: Maybe.withDefault ?

File: src/Update/Main.elm [hunk 5]
```diff
@@ -38,54 +37,77 @@ update : Msg -> Model -> ( Model, Cmd Msg )
...
 
-        UrlChange location ->
+        UrlChanged location ->
             Update.Route.update { model | location = location }
 
         Back ->
-            model ! [ Navigation.back 1 ]
+            ( model
+            , Cmd.none
+              -- @TODO: add it again
```
Comment (Natim) [inline src/Update/Main.elm]: Navigation back and forth.

File: src/View/App.elm (hunk 4 above)
Comment (vjousse) [reply src/View/App.elm:130]: Nope, we don't want to display the confirm view if `confirm` is Nothing

File: src/Util.elm (hunk 3 above)
Comment (vjousse) [reply src/Util.elm:25]: Fixed using QS package. `Url.Parser.Query` is more useful for parsing full URLs

**Pull request merged**